src/alchemlyb/workflows/abfe.py

```diff
@@ -441,7 +444,12 @@ def estimate(self, estimators=("MBAR", "BAR", "TI"), **kwargs):
         for estimator in estimators:
             if estimator == "MBAR":
                 self.logger.info("Run MBAR estimator.")
-                self.estimator[estimator] = MBAR(**kwargs).fit(u_nk)
+                self.logger.info(
+                    "By default, n_bootstraps=50 is used to estimate the "
+                    "MBAR error. Supply n_bootstraps=0 to use analytic error."
+                )
+                estimator_kwargs = {"n_bootstraps": 50, **kwargs}
```

Comment:
This will fail when I actually supply `n_bootstraps` as a kwarg because then the dict contains the same key twice ... a test would have caught it.

Instead, I would do
```python
# use 0 as default to remain backwards-compatible
# change in release 2.3
kwargs.setdefault('n_bootstraps', 0)
...
...
if estimator == "MBAR":
  estimator_kwargs = kwargs
  self.logger.info("Run MBAR estimator")
  self.logger.info("Estimating MBAR error with n_bootstraps={n_bootstraps} (0 is analytical estimate)", estimator_kwargs)
elif estimator == "BAR":
  estimator_kwargs = kwargs.copy()
  estimator_kwargs.pop('n_bootstraps', None)
  ...
  self.estimator[estimator] = BAR(**kwargs).fit(u_nk)
```
i.e., modify the copy of kwargs as needed but keep everything in kwargs.

Reply:
Sorry, what do you mean this will fail?

```
>>> kwargs = {"n_bootstraps": 0}
>>> estimator_kwargs = {"n_bootstraps": 50, **kwargs}
>>> print(estimator_kwargs)
{'n_bootstraps': 0}
```

Gives the expected behaviour where kwargs overrides the default in estimator_kwargs?

Reply:
You're right, I am a bit surprised. (I was in the process of trying it out but then wrote the review comment without finishing my testing.)

My concern remains that the default ought to be n_bootstraps=0 to initially have the same behavior.